static/js/hotkey.js

```diff
@@ -476,7 +477,16 @@ exports.process_hotkey = function (e, hotkey) {
         case "vim_right":
         case "tab":
         case "shift_tab":
-            if (overlays.recent_topics_open()) {
+        case "open_recent_topics":
+        case "escape":
+            if (
+                ["#recent_topics", "#", ""].includes(window.location.hash) &&
```

Comment:
We should abstract this first check to a function with a name exported by `hashchange.js`.

Reply:
Okay, Opened #17330 .